domain/schema/controller.go

```diff
@@ -591,3 +592,42 @@ CREATE TABLE flag (
 );
 `)
 }
+
+// The permissions schema
+func permissionsSchema() schema.Patch {
+	return schema.MakePatch(`
+CREATE TABLE access_type (
+    id     INT PRIMARY KEY,
+    type   Text
+);
+
+CREATE UNIQUE INDEX idx_access_type
+ON access_type (type);
+
+-- Maps to the Access type in core/permission package.
+INSERT INTO access_type VALUES
```

Comment:
Access types shouldn't refer to entities directly (such as add-model). There should be another table for entities, and a join table that forms a cross-product of valid entity-access pairs.

Reply:
`add-model` is an access type, not an entity. `add-model` is an access type for clouds. https://github.com/juju/juju/blob/62efec59b2402498bdc2f28b77d3dc8aeabadcc4/core/permission/access.go

It just happens that we have model level accesses as well with different access types. Calling it `add` for the cloud entities could imply that the user can add clouds.